v3/plugins/redhat/dependency-analytics/0.0.13/meta.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: v2
+publisher: redhat
+name: dependency-analytics
+version: 0.0.13
+type: VS Code extension
+displayName: Dependency Analytics
+title: Insights about your application dependencies
+description: Insights about your application dependencies, Security, License compatibility and AI based guidance to choose appropriate dependencies for your application.
+icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
+repository: https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension
+category: Other
+firstPublicationDate: '2018-10-03'
+spec:
+  containers:
+    - image: "docker.io/eclipse/che-remote-plugin-dependency-analytics-0.0.13:next"
+      memoryLimit: "512Mi"
```

Comment:
Is `512Mi` really necessary for this plugin? Such a high default value means it cannot easily be used in many workspaces.

Reply:
I thought `memoryLimit` caps the max RAM usage. Does it reserves the given amount of memory?

Reply:
On the OpenShift side, it is a cap (with request as the lower bound), but on clusters that aren't configured to overprovision memory, they still eat into quotas. 

E.g. on Hosted Che, the maximum memory limit for a Che workspace is `3Gi`

Reply:
I tried reducing to 256Mi, it is failing on some big maven stacks like [che](https://github.com/eclipse/che). I see the following error,
```
Initial heap size set to a larger value than the maximum heap size
```

Reply:
I tried to use the `requests` attribute, it didn't seem to be working.

```
  containers:
    - image: "docker.io/eclipse/che-remote-plugin-dependency-analytics-0.0.12:7.1.0"
      resources:
        requests:
          memory: "128Mi"
        limits:
          memory: "512Mi"
```

Reply:
I don't think requests are taken into account at the moment.

Reply:
If 256 is too small and 512 is too big, what about 384?

Reply:
Memory consumption is based on the stack which we scan. Stacks like [che](https://github.com/eclipse/che) consumes almost 500MB while doing the scan. Anyways, `memoryLimit` can be overridden from the devfile., which gives more control to the users to reduce/increase the value based on the stack they scan. I'm fine with either 384 or 512.